hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSourceManager.java

```diff
@@ -579,8 +579,8 @@ private void interruptOrAbortWhenFail(ReplicationQueueOperation op) {
       if (e.getCause() != null && e.getCause() instanceof KeeperException.SystemErrorException
```

Comment:
I tried rewriting this big nasty if-expression using an `Optional`. Tell me what you think

```java
      if (Optional.of(e.getCause())
        .filter(c -> c instanceof KeeperException.SystemErrorException)
        .map(c -> (KeeperException.SystemErrorException) c)
        .map(Exception::getCause)
        .filter(c -> c instanceof InterruptedException)
        .isPresent()) {
```

Reply:
missed this comment, will me get back.